ui/src/dashboards/utils/tempVars.ts

```diff
@@ -140,3 +139,74 @@ export const findInvalidTempVarsInURLQuery = (
 
   return urlQueryParamsTempVarsWithInvalidValues
 }
+
+const makeSelected = (template: Template, value: string): Template => {
+  const found = template.values.find(v => v.value === value)
+
+  let valueToChoose
+  if (found) {
+    valueToChoose = found.value
+  } else {
+    valueToChoose = getDeep<string>(template, 'values.0.value', '')
+  }
+
+  const valuesWithSelected = template.values.map(v => {
+    if (v.value === valueToChoose) {
+      return {...v, selected: true}
+    } else {
+      return {...v, selected: false}
+    }
+  })
+
+  return {...template, values: valuesWithSelected}
+}
+
+export const pickSelected = (template: Template): Template => {
+  const selectedValue = ''
+
+  return makePicked(template, selectedValue)
+}
+
+export const makePicked = (template: Template, value: string): Template => {
+  const found = template.values.find(v => v.value === value)
+  const selectedValue = template.values.find(v => v.selected)
+
+  let valueToChoose: string
+  if (found) {
+    valueToChoose = found.value
+  } else if (selectedValue) {
+    valueToChoose = selectedValue.value
+  } else {
+    valueToChoose = getDeep<string>(template, 'values.0.value', '')
+  }
+
+  const valuesWithPicked = template.values.map(v => {
+    if (v.value === valueToChoose) {
+      return {...v, picked: true}
+    } else {
+      return {...v, picked: false}
+    }
+  })
+
+  return {...template, values: valuesWithPicked}
+}
+
+export const reconcileSelectedAndPickedValues = (
+  nextTemplate: Template,
+  nextNextTemplate: Template
+): Template => {
+  const pickedValue = nextTemplate.values.find(v => v.picked)
+  const selectedValue = nextTemplate.values.find(v => v.selected)
+  // make picked from selected
+  const TemplateWithPicked = makeSelected(
+    nextNextTemplate,
+    getDeep<string>(selectedValue, 'value', '')
+  )
+
+  const TemplateWithPickedAndSelected = makePicked(
```

Comment:
this doesn't appear to be a type -- is it capitalized for a reason? if not, could you de-capitalize it for consistency?